src/rs/lib.rs

```diff
@@ -651,8 +651,16 @@ macro_rules! define_quic_handle_impl {
             }
 
             /// Closes the handle and consumes it.
-            pub fn close(self) {
+            pub fn close(mut self) {
+                // close_inner implementation for listener, connection and stream
+                // is suppose to cleanup the context associated with the handle.
+                // Handle context is dropped after handle ffi close  call completes to ensure that
+                // it outlives the callbacks.
+                // context raw pointer is not set to null after ffi close call because a closed
+                // handle should not be accessed anymore.
                 self.close_inner();
+                // prevent drop to call ffi function again.
+                self.handle = std::ptr::null_mut();
```

Comment:
```suggestion
                // The close_inner implementation for listener, connection and stream
                // is suppose to cleanup the context associated with the handle.
                // Handle context is dropped after handle ffi close call completes to ensure that
                // it outlives the callbacks.
                // Context raw pointer is not set to null after ffi close call because a closed
                // handle should not be accessed anymore.
                self.close_inner();
                // Prevent drop to call ffi function again.
                self.handle = std::ptr::null_mut();
```